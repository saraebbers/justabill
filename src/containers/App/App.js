import React, { Component } from 'react';
import { Route, Switch, Link, NavLink } from 'react-router-dom';
import '../../index.scss';
import Welcome from '../../components/Welcome/Welcome';
import Display from '../../components/Display/Display';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchBillsThunk } from '../../thunks/fetchBillsThunk';
import { withRouter } from 'react-router';



export class App extends Component {

  async componentDidMount() {
    const url = 'https://api.propublica.org/congress/v1/115/both/bills/enacted.json'
    await this.props.fetchBillsThunk(url)
  }

  render() {
    return (
      <div >
        <section className="App">
          <Link to='/' className='header'>
            <header>
              <h1>Just A Bill</h1>
            </header>
          </Link>
          <div className='nav-section'>
            <NavLink to='/bills' className='nav'>Bills</NavLink>
            <NavLink to='/favorites' className='nav'>Favorites</NavLink>
          </div>
        </section>
        <Switch>
          <Route exact path='/' component={Welcome} />
          <Route exact path='/bills' render={() => <Display type="Bills" />} />
          <Route exact path='/favorites' render={() => <Display type="Favorites" />} />
          <Route render={() => (
            <div className='redirect'>
              <Link to='/' className='lnk-redirect'>
                <h1 >OOPS!  Can't find that page</h1>
                <button className='home'> Click Here To Return Home </button>
              </Link>
            </div>
            )} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  billArray: state.bills,
  isLoading: state.isLoading,
  errorMessage: state.errorMessage,
})

export const mapDispatchToProps = (dispatch) => ({
  fetchBillsThunk: (url) => dispatch(fetchBillsThunk(url)),
})

App.propTypes = {
  billArray: PropTypes.array.isRequired,
  isLoading: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string.isRequired,
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
