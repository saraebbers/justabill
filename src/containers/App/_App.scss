* {
  margin: 0;
  font-family: 'Open Sans', sans-serif;
}

.App {
  height: 100vh;
  background: url(./assets/billOfRights.png)
    no-repeat
    top center
}

h1 {
  text-align: center;
  font-size: 5.5rem;
  color: #420C14;
  padding: 50px;
  text-shadow: 10px 10px 10px #DBCFB0;
  font-family: 'Philosopher', sans-serif;

    &:hover {
    color: #718F94;
  }
}

a {
  text-decoration: none;
  cursor: pointer;
}

.nav-section {
  background-color: #420C14;
  font-size: 1.25rem;
  font-family: 'Open Sans';
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 5px;
  justify-items: center;
  justify-content: space-around;
}

.nav {
  background-color: #DBCFB0;
  margin: 15px 5px;
  padding: 10px;
  border: solid #718F94 2px;
  color: #420C14;

  &.active {
    background-color: #718F94;
  }

  &:hover {
    background-color: #ADA380;
    color: #191970;
  }
}