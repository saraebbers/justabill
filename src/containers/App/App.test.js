import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App Container', () => {
  it('should match the screenshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})