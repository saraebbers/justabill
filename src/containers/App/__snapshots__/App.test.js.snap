// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="App"
        >
          <Link
            className="header"
            replace={false}
            to="/"
          >
            <header>
              <h1>
                Just A Bill
              </h1>
            </header>
          </Link>
          <div
            className="nav-section"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/bills"
            >
              Bills
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/favorites"
            >
              Favorites
            </NavLink>
          </div>
        </section>,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            exact={true}
            path="/bills"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />
          <Route
            render={[Function]}
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="header"
              replace={false}
              to="/"
            >
              <header>
                <h1>
                  Just A Bill
                </h1>
              </header>
            </Link>,
            <div
              className="nav-section"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/bills"
              >
                Bills
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/favorites"
              >
                Favorites
              </NavLink>
            </div>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <header>
                <h1>
                  Just A Bill
                </h1>
              </header>,
              "className": "header",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Just A Bill
                </h1>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Just A Bill",
                },
                "ref": null,
                "rendered": "Just A Bill",
                "type": "h1",
              },
              "type": "header",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/bills"
                >
                  Bills
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/favorites"
                >
                  Favorites
                </NavLink>,
              ],
              "className": "nav-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Bills",
                  "className": "nav",
                  "to": "/bills",
                },
                "ref": null,
                "rendered": "Bills",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Favorites",
                  "className": "nav",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              exact={true}
              path="/bills"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />,
            <Route
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/bills",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="App"
          >
            <Link
              className="header"
              replace={false}
              to="/"
            >
              <header>
                <h1>
                  Just A Bill
                </h1>
              </header>
            </Link>
            <div
              className="nav-section"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/bills"
              >
                Bills
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav"
                to="/favorites"
              >
                Favorites
              </NavLink>
            </div>
          </section>,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              exact={true}
              path="/bills"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />
            <Route
              render={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="header"
                replace={false}
                to="/"
              >
                <header>
                  <h1>
                    Just A Bill
                  </h1>
                </header>
              </Link>,
              <div
                className="nav-section"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/bills"
                >
                  Bills
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav"
                  to="/favorites"
                >
                  Favorites
                </NavLink>
              </div>,
            ],
            "className": "App",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <header>
                  <h1>
                    Just A Bill
                  </h1>
                </header>,
                "className": "header",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    Just A Bill
                  </h1>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Just A Bill",
                  },
                  "ref": null,
                  "rendered": "Just A Bill",
                  "type": "h1",
                },
                "type": "header",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav"
                    to="/bills"
                  >
                    Bills
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav"
                    to="/favorites"
                  >
                    Favorites
                  </NavLink>,
                ],
                "className": "nav-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Bills",
                    "className": "nav",
                    "to": "/bills",
                  },
                  "ref": null,
                  "rendered": "Bills",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Favorites",
                    "className": "nav",
                    "to": "/favorites",
                  },
                  "ref": null,
                  "rendered": "Favorites",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                exact={true}
                path="/bills"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
              />,
              <Route
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/bills",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
